### ISSUE TITLE
Ranksystem Losing connection after a while 

### ISSUE THREAD

**Issue by Roshke** [Newcomer1989/TSN-Ranksystem#791]
Description | Fill out below
-- | --
Ranksystem Version | 1.3.14-beta-2 - also tried the Stable it started after the latest Release Update
Are you using TeamSpeak or TeaSpeak? | TeamSpeak 3.13.2 on Linux
Link to your Ranksystem Stats page | https://teamspeak.rs/ranks/stats/
Pastebin Link to your Ranksystem Log *¹ |  https://pastebin.com/5CDzecAN
---

### 1) What did you do?
Nothing, after latest Stable Update the issues started to happen that I get random disconnects and then it never connects.
Web Server is communicating locally - ts3.roshke.me Resolves to 10.0.0.2 (internal DNS) and both Public and Private IPv4 of the Web Server are in Query Whitelist, it worked before nothing was changed.

### 2) What result did you get?
Bot disconnects randomly and never connects until someone trough web interface restarts it, the cronjob for the "worker.php check" says nothing, "worker.php status" says it's running but it's not connected.
I will try now to use RAW 10011 Port

### 3) What result did you expect?

---

#### *¹ Ranksystem Log - HOW-TO
- Change the Log-Level to `6 - Debug` (-> Webinterface -> Other -> Log-Level).
- Change the default language to `EN english` (-> Webinterface -> Other -> Language).
- Restart the Ranksystem Bot and let it run for at least 10 minutes (-> Webinterface -> Start / Stop bot -> Restart bot).
- Add a Ranksystem-Log extract which contains all since the last restart (-> Webinterface -> Start / Stop bot -> at bottom).
- Please use https://pastebin.com/ for sharing the log.

**Comment by Newcomer1989** [comment]
It would be helpful to get a Ranksystem Log with the Log-Level "6 - Debug".

```
2020-11-28 21:06:20.164520   NOTICE    2: fgets(): Failure 'Timed out waiting on socket' (-9) on line 149 in /var/www/teamspeak.rs/data/www/teamspeak.rs/ranks/libs/ts3_lib/Transport/TCP.php
2020-11-28 20:59:14.586425   INFO      Download avatar: /avatar_ibkjkabefobbejeeomkepdjnelcpkhnildhiolip

2020-11-28 16:35:58.868522   NOTICE    2: fgets(): Failure 'Timed out waiting on socket' (-9) on line 149 in /var/www/teamspeak.rs/data/www/teamspeak.rs/ranks/libs/ts3_lib/Transport/TCP.php
2020-11-28 16:34:34.211672   INFO      Download avatar: /avatar_ibkjkabefobbejeeomkepdjnelcpkhnildhiolip


2020-11-29 00:00:55.252536   NOTICE    2: fgets(): Failure 'Timed out waiting on socket' (-9) on line 149 in /var/www/teamspeak.rs/data/www/teamspeak.rs/ranks/libs/ts3_lib/Transport/TCP.php
2020-11-28 23:57:03.172322   INFO      Download avatar: /avatar_ekiclbdnokkjjmkpamfkahgcbhbadkcocgglehmh
```

For now it seems to be, you are running in timeouts by downloading avatar files.

What is the configuration value of your avatar-delay (Webinterface -> TeamSpeak -> Avatar Delay)?

**Comment by Fubister** [comment]
I can confirm this behaviour. Running the latest stable version. Once bot loses the connection to server it never reconnects despite trying, the only solution is to restart the bot. Some debug logs: https://pastebin.com/wcr4xRQU

**Comment by Newcomer1989** [comment]
There is a current beta out, which solves the problem -> 1.3.14-beta-3

**Comment by Newcomer1989** [comment]
Should be solved with the current release 1.3.14.
If not, please feel free to reopen this issue.
